src/_tests/integration/integration-utils.js

```diff
@@ -1,5 +1,40 @@
+const waitForFn = async ({ fn, interval = 2000, timeout = 10000 }) => {
+  const readyState = new Promise(resolve => {
+    const start = Date.now()
+    const intervalId = setInterval(() => {
```

Comment:
This will wait a minimum of 2 seconds before trying the function. Should we make one call up front to see if it's already successful?